src/plugins/intel_gpu/src/plugin/infer_request.cpp

```diff
@@ -235,7 +235,19 @@ void InferRequest::set_input(const std::string& name, const Blob::Ptr& data) {
     } else {
         if (data->buffer() == nullptr)
             IE_THROW(NotAllocated) << str_input_not_allocated << " Input name: \'" << name << "\'";
-        _inputs[name] = data;
+        if (_inputs.find(name) != _inputs.end() && _inputs[name]->byteSize() == data->byteSize()) {
```

Comment:
We need to save user tensor, so the expected behavior is `_inputs[name] = data;` and then enqueue copy from user blob (system memory) to USM device blob (w/o lock)

Reply:
I reverted the change and updated to allocate lockable memory for `_deviceInputs`. This makes it to store user tensors and treat them the same as other data types in `prepare_inputs()`. Thank you.